src/PersonObjects/PersonMethods.ts

```diff
@@ -44,7 +44,8 @@ export function gainDefenseExp(this: Person, exp: number): void {
   }
 
   this.skills.defense = calculateSkill(this.exp.defense, this.mults.defense * currentNodeMults.DefenseLevelMultiplier);
-  const ratio = this.hp.current / this.hp.max;
+  //In case MaxHP is Infinite, setting ratio to 1
+  const ratio = this.hp.current / this.hp.max || 1;
```

Comment:
This doesn't work, it sets the ratio to 1 when it should be 0.

This is a case of protecting against an Infinity where (IMO) there shouldn't be one (health should not be infinite), *but* I'm OK with defense in depth so do it like this:
```suggestion
  const ratio = Number.isFinite(this.hp.current) ? this.hp.current / this.hp.max : 1;
```
That will also catch NaNs in the current hp.